core/classes/IssueStatusChangeTimelineEvent.class.php

```diff
@@ -45,24 +55,67 @@ public function __construct( $p_timestamp, $p_user_id, $p_issue_id, $p_old_statu
 		$this->issue_id = $p_issue_id;
 		$this->old_status = $p_old_status;
 		$this->new_status = $p_new_status;
+		$this->type = $this->change_type();
 	}
 
 	/**
-	 * Returns html string to display
-	 * @return string
+	 * Return the type of status change
+	 * @return int One of the status change constants defined above
 	 */
-	public function html() {
+	private function change_type() {
 		$t_resolved = config_get( 'bug_resolved_status_threshold' );
 		$t_closed = config_get( 'bug_closed_status_threshold' );
 
-		if( $this->old_status < $t_closed && $this->new_status >= $t_closed ) {
-			$t_string = sprintf( lang_get( 'timeline_issue_closed' ), user_get_name( $this->user_id ), string_get_bug_view_link( $this->issue_id ) );
-		} else if( $this->old_status < $t_resolved && $this->new_status >= $t_resolved ) {
-			$t_string = sprintf( lang_get( 'timeline_issue_resolved' ), user_get_name( $this->user_id ), string_get_bug_view_link( $this->issue_id ) );
+		if( $this->old_status < $t_resolved && $this->new_status >= $t_resolved ) {
+			return IssueStatusChangeTimelineEvent::RESOLVED;
+		} else if( $this->old_status < $t_closed && $this->new_status >= $t_closed ) {
+			return IssueStatusChangeTimelineEvent::CLOSED;
 		} else if( $this->old_status >= $t_resolved && $this->new_status < $t_resolved ) {
-			$t_string = sprintf( lang_get( 'timeline_issue_reopened' ), user_get_name( $this->user_id ), string_get_bug_view_link( $this->issue_id ) );
+			return IssueStatusChangeTimelineEvent::REOPENED;
 		} else {
-			return '';
+			return IssueStatusChangeTimelineEvent::IGNORED;
+		}
+	}
+
+	/**
+	 * Whether to skip this timeline event.
+	 * This normally implements access checks for the event.
+	 * @return boolean
+	 */
+	public function skip() {
+		return $this->type == IssueStatusChangeTimelineEvent::IGNORED;
+	}
+
+	/**
+	 * Returns html string to display
+	 * @return string
+	 */
+	public function html() {
+		switch( $this->type ) {
+			case IssueStatusChangeTimelineEvent::RESOLVED:
+				$t_string = sprintf(
+					lang_get( 'timeline_issue_resolved' ),
+					user_get_name( $this->user_id ),
+					string_get_bug_view_link( $this->issue_id )
+				);
+				break;
+			case IssueStatusChangeTimelineEvent::CLOSED:
+				$t_string = sprintf(
+					lang_get( 'timeline_issue_closed' ),
+					user_get_name( $this->user_id ),
+					string_get_bug_view_link( $this->issue_id )
+				);
+				break;
+			case IssueStatusChangeTimelineEvent::REOPENED:
+				$t_string = sprintf(
+					lang_get( 'timeline_issue_reopened' ),
+					user_get_name( $this->user_id ),
+					string_get_bug_view_link( $this->issue_id )
+				);
+				break;
+			case IssueStatusChangeTimelineEvent::IGNORED:
+			default:
```

Comment:
I don't think so, based on my testing, but maybe I missed something ?

The logic for excluding event (i.e. using the skip() method) is actually in timeline API's timeline_events() function; the array it generates does (or should) not contain any "empty" events (IGNORED type), so in theory we should never reach this default case unless something went wrong. 

It's belt and braces I guess - would you prefer that I remove lines 116-118 ?